pkg/apis/v1alpha2/crd_validation_cel_test.go

```diff
@@ -59,6 +59,42 @@ var _ = Describe("CEL/Validation", func() {
 			},
 		}
 	})
+	Context("VnetSubnetID", func() {
+		DescribeTable("Should only accept valid VnetSubnetID", func(vnetSubnetID string, expected bool) {
+			nodeClass := &v1alpha2.AKSNodeClass{
+				ObjectMeta: metav1.ObjectMeta{Name: strings.ToLower(randomdata.SillyName())},
+				Spec: v1alpha2.AKSNodeClassSpec{
+					VNETSubnetID: &vnetSubnetID,
+				},
+			}
+			if expected {
+				Expect(env.Client.Create(ctx, nodeClass)).To(Succeed())
+			} else {
+				Expect(env.Client.Create(ctx, nodeClass)).ToNot(Succeed())
+			}
+		},
+			Entry("valid VnetSubnetID", "/subscriptions/12345678-1234-1234-1234-123456789012/resourceGroups/rgname/providers/Microsoft.Network/virtualNetworks/vnet/subnets/subnet", true),
```

Comment:
Can add in the edge cases like we talked about for 1 character subnets, etc. 

I'd personally add in some of the non letter special characters just to ensure those work.

However, all this depends on how deep you want to go into the validation part, and like we talked about might be better to backoff and make less restrictive validation here instead.

Reply:
Oh, I see you did add those in which is great. My assumption here was that all the valid cases would come first, followed by the invalid ones. I think a light refactor to just block them out as the complete valid set of cases, and then invalid would be nice.